Automattic/jetpack
# Tests: Speed up linting with php-parallel-lint

#### Changes proposed in this Pull Request:

We currently run `php -l` over most of the PHP files by having a PHPUnit
test that shells out to `find`. Which works, but it's pretty slow (about
35 seconds in my local environment).

We can improve it significantly using [php-parallel-lint](https://packagist.org/packages/php-parallel-lint/php-parallel-lint). Locally for
me it's down to 2.8 seconds.

I've also moved the running of it out of PHPUnit and into a separate
script. I'd have added a composer alias for the new script too, but
"php:lint" was already taken for phpcs.

#### Jetpack product discussion


None.

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* See if Travis tests pass, and that the logs for the builds with `PHP_LINT=1` show running of the linter.
* Try running `bin/parallel-lint.sh` locally.
* Try adding a lint error, then running `bin/parallel-lint.sh` locally.

#### Proposed changelog entry for your changes:

* Developer only, none needed.
